src/audio/component.c

```diff
@@ -505,3 +505,47 @@ int comp_copy(struct comp_dev *dev)
 
 	return ret;
 }
+
+#if CONFIG_IPC_MAJOR_4
+static uint32_t get_sample_group_size_in_bytes(const struct ipc4_audio_format fmt)
+{
+	return (fmt.depth >> 3) * fmt.channels_count;
+}
+
+static uint32_t get_one_ms_in_bytes(const struct ipc4_audio_format fmt)
+{
+	/* TODO Reference Firmware also has systick multiplier and divider in this equation */
+	return get_sample_group_size_in_bytes(fmt) *
+			(SOF_DIV_ROUND_UP(fmt.sampling_frequency, 1000) / 1000);
```

Comment:
don't think you need to divide by 1000 now any more, `SOF_DIV_ROUND_UP()` already does it. And after you've removed it please also drop parentheses

Reply:
The divide is part of the equation, it's correct

Reply:
Here's the difference to your previous version:
![Screenshot_20240704_083619](https://github.com/thesofproject/sof/assets/1363683/71c400d7-366a-440f-ac91-d6f1c639f1a2)
i.e. basically you replaced `(fmt.sampling_frequency + 999) / 1000` with `SOF_DIV_ROUND_UP(fmt.sampling_frequency, 1000) / 1000` and that is wrong, because `SOF_DIV_ROUND_UP()` is defined as
```
#define SOF_DIV_ROUND_UP(val, div) (((val) + (div) - 1) / (div))
```
and therefore already performs the division by 1000 for you. So, either your original formula was wrong and you actually wanted to divide by 1000000 and not 1000, or this your update is wrong.

Reply:
The original formula was wrong, yes. I should've mentioned it, sorry.

Reply:
I think @tobonex the formula is not correct now:
- 16bit stereo 48000Hz example -> sample group size in bytes 4
- this function would return now: 4 * (48000 / 1000) / 1000 = 0.912 = 0

So I think this should be:
```
return get_sample_group_size_in_bytes(fmt) *
			SOF_DIV_ROUND_UP(fmt.sampling_frequency, 1000);
```

Which gives you 4 * 48000 / 1000 = 192 bytes (which is correct. I think you missed SOF_DIV_ROUND_UP does actually do the dvision.

Reply:
Ah, ok i get it now, ROUND_UP_DIV adds division COMPARED to ROUND_UP. I'll fix.